### PR TITLE
SAK-43760 Rubrics: Can not use keyboard-only to edit criterion or rating

### PR DESCRIPTION
**Pull request by MRutea**
SAK-43760 Rubrics: Can not use keyboard-only to edit criterion or rating

SAK-43760 Rubrics: Can not use keyboard-only to edit criterion or rating

### CODE REVIEW

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criteria.js [hunk 1]
```diff
@@ -419,6 +419,13 @@ export class SakaiRubricCriteria extends RubricsElement {
     .done(() => this.letShareKnow())
     .fail((jqXHR, error, message) => {console.log(error); console.log(message);});
   }
+
+ openEditWithKeyboard(e){
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criteria.js:423]: Can you stick to the formatting in the rest of the file? Space before the opening brace and a blank line to start the function body, unless the function is only one line of code.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criteria.js [hunk 2]
```diff
@@ -419,6 +419,13 @@ export class SakaiRubricCriteria extends RubricsElement {
...
+    if(e.keyCode == 32 || e.keyCode == 32 ){
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criteria.js:424]: Did you mean to check two codes here? And, could you leave a space after the if, and one before the opening brace?

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criteria.js [hunk 3]
```diff
@@ -419,6 +419,13 @@ export class SakaiRubricCriteria extends RubricsElement {
...
+      this.cloneCriterion(e)
+    }
+  }
+
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criteria.js:428]: Could you take this blank line out?

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criterion-edit.js [hunk 4]
```diff
@@ -163,6 +163,13 @@ export class SakaiRubricCriterionEdit extends RubricsElement {
     this.dispatchEvent(new CustomEvent('criterion-edited', {detail: {id: data.id, title: data.title, description: data.description}}));
     this.dispatchEvent(new SharingChangeEvent());
   }
+
+ openEditWithKeyboard(e){
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criterion-edit.js:167]: Could you format this function in line with the rest of the file? And, I think you meant to test two different codes here.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criterion-rating-edit.js [hunk 5]
```diff
@@ -145,6 +145,13 @@ export class SakaiRubricCriterionRatingEdit extends RubricsElement {
     this.dispatchEvent(new CustomEvent('delete-rating', {detail: this.rating}));
     this.hideToolTip();
   }
+
+ openEditWithKeyboard(e){
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric-criterion-rating-edit.js:149]: Please format this like the rest of the file. And, you're checking for 32 twice.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js [hunk 6]
```diff
@@ -329,6 +329,27 @@ export class SakaiRubric extends RubricsElement {
       sakaiItemDelete.requestUpdate("rubric", this.rubric);
     }
   }
+
+ openEditWithKeyboard(e){
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:333]: Can you format this like the rest of the file, please? Also, you're checking for the same keycode twice.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js [hunk 7]
```diff
@@ -329,6 +329,27 @@ export class SakaiRubric extends RubricsElement {
...
+    const classType = e.target.className;
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:334]: use classlist for this:
https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js [hunk 8]
```diff
@@ -329,6 +329,27 @@ export class SakaiRubric extends RubricsElement {
...
+    const weighted = 'weighted';
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:335]: Why bother with a const for a string you're only going to use once? Just use the string.

File: webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js [hunk 9]
```diff
@@ -329,6 +329,27 @@ export class SakaiRubric extends RubricsElement {
...
+    const share = 'share';
+    const clone = 'clone';
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/js/rubrics/sakai-rubric.js:337]: Same with all these variables. You don't need them.

**Review by adrianfish — CHANGES_REQUESTED** [review]

**Comment by MRutea** [comment]
@adrianfish 
Hello Adrian.
Thanks again for your comments.
I will be more careful with those details.
I have made the corrections that you suggest.

**Comment by adrianfish** [comment]
@MRutea I'm a bit pedantic about the formatting, sorry! I'm really glad you're working on these rubrics tickets.

**Review by adrianfish — APPROVED** [review]
GTG. Thanks for making those changes.

**Pull request merged**